scripts/build/builders/bouffalolab.py

```diff
@@ -114,30 +123,65 @@ def __init__(self,
         self.argsOpt.append('board=\"{}\"'.format(self.board.GnArgName()))
         self.argsOpt.append('baudrate=\"{}\"'.format(baudrate))
 
+        if not (enable_wifi or enable_thread or enable_ethernet):
+            # decide default connectivity for each chip
+            if bouffalo_chip == "bl602":
+                enable_wifi, enable_thread, enable_ethernet = True, False, False
+            elif bouffalo_chip == "bl702":
+                enable_wifi, enable_thread, enable_ethernet = False, True, False
+            elif bouffalo_chip == "bl702l":
+                enable_wifi, enable_thread, enable_ethernet = False, True, False
+
         if bouffalo_chip == "bl602":
-            self.argsOpt.append('chip_enable_openthread=false')
-            self.argsOpt.append('chip_enable_wifi=true')
-        if bouffalo_chip == "bl702":
+
+            if enable_ethernet or enable_thread:
+                raise Exception('SoC %s doesn\'t support connectivity Ethernet/Thread.' % bouffalo_chip)
+
+        elif bouffalo_chip == "bl702":
+
             self.argsOpt.append('module_type=\"{}\"'.format(module_type))
-            if board == BouffalolabBoard.BL706_ETH:
-                self.argsOpt.append('chip_config_network_layer_ble=false')
-                self.argsOpt.append('chip_enable_openthread=false')
-                self.argsOpt.append('chip_enable_wifi=false')
-            elif board == BouffalolabBoard.BL706_WIFI:
-                self.argsOpt.append('chip_enable_openthread=false')
-                self.argsOpt.append('chip_enable_wifi=true')
-            else:
-                self.argsOpt.append('chip_enable_openthread=true')
-                self.argsOpt.append('chip_enable_wifi=false')
+            if board != BouffalolabBoard.BL706DK:
+                if enable_ethernet or enable_wifi:
+                    raise Exception('Board %s doesn\'t support connectivity Ethernet/Wi-Fi.' % board)
+
         elif bouffalo_chip == "bl702l":
-            self.argsOpt.append('chip_enable_openthread=true')
+
+            if enable_ethernet or enable_wifi:
+                raise Exception('SoC %s doesn\'t support connectivity Ethernet/Wi-Fi currently.' % bouffalo_chip)
+
+        if enable_wifi or enable_thread:
+            self.argsOpt.append('chip_config_network_layer_ble=true')
+        else:
+            # for enable_ethernet, do not need ble for commissioning
+            self.argsOpt.append('chip_config_network_layer_ble=false')
+
+        if enable_ethernet:
+            self.argsOpt.append('chip_enable_ethernet=true')
+            self.argsOpt.append('chip_mdns="minimal"')
```

Comment:
if both enable_ethernet and enable_wifi are set, this sets the arg opts duplicated.

Should we have rules like:
  - you cannot have thread and wifi/ethernet at the same time (since then mdns is unclear)
  - mdns and ipv4 rules should be based wifi or ethernet instead of independently re-added for each

Also note that ipv4 is not required for wifi/ethernet based on the spec. Other build systems have ipv6only targets to validate that things work in a ipv6only environment (I believe certification should test that)

Reply:
Duplicated options for enable_ethernet and enable_wifi are moved.

For options `chip_inet_config_enable_ipv4` and `chip_mdns`, 
- Normally, default setting of `chip_inet_config_enable_ipv4` puts in args.gni of each platform
    - most Wi-Fi/ethernet platform set it to true for the scenario in IPv4 only AP
    - and we will add `-ipv6only` to build ipv6 only application in future 
- Normally, default setting  of `chip_mdns` is set in `src/platform/device.gni`

The reason,  we extract these two options to `bouffalolab.py`, is why we have a developing BL706+BL602 Wi-Fi /BL706 Ethernet Matter device with openthread border router integrated and it has more flexibility to config for Matter + OTBR mixed device and standalone Wi-Fi/Ethernet Matter device.